server/routes/api.js

```diff
@@ -1073,129 +1089,359 @@ router.post('/capcodeImport', isLoggedIn, function (req, res, next) {
+        var givenname = req.body.givenname;
+        var surname = req.body.surname || '';
+        var email = req.body.email;
+        var password = req.body.newpassword || null;
+        var role = req.body.role || 'user';
+        var status = req.body.status || 'disabled';
+        var lastlogondate = Date.now()
+        console.time('insert');
+        db.from('users')
+          .returning('id')
+          .where('id', '=', id)
+          .modify(function (queryBuilder) {
+            if (password != null) {
+              const salt = bcrypt.genSaltSync();
+              const hash = bcrypt.hashSync(password, salt);
+              if (id == null) {
+                queryBuilder.insert({
+                  id: id,
+                  username: username,
+                  givenname: givenname,
+                  surname: surname,
+                  password: hash,
+                  email: email,
+                  role: role,
+                  status: status,
+                  lastlogondate: lastlogondate
+                })
+              } else {
+                queryBuilder.update({
+                  id: id,
+                  username: username,
+                  givenname: givenname,
+                  surname: surname,
+                  password: hash,
+                  email: email,
+                  role: role,
+                  status: status,
+                  lastlogondate: lastlogondate
+                })
+              }
+            } else {
+              queryBuilder.update({
+                id: id,
+                username: username,
+                givenname: givenname,
+                surname: surname,
+                email: email,
+                role: role,
+                status: status,
+                lastlogondate: lastlogondate
+              })
+
+            }
+          })
+          .then((result) => {
+            console.timeEnd('insert');
+            res.status(200).send({ 'status': 'ok', 'id': result })
+          })
+          .catch((err) => {
+            console.timeEnd('insert');
+            logger.main.error(err)
+            res.status(500).send(err);
+          })
+        logger.main.debug(util.format('%o', req.body || 'request body empty'));
+      } else {
+        res.status(500).json({ message: 'Error - required field missing' });
+      }
+    }
+  })
+  .delete(isAdmin, function (req, res, next) {
+    var id = parseInt(req.params.id, 10);
+    if (id != 1) {
+      logger.main.info('Deleting User ' + id);
+      db.from('users')
+        .del()
+        .where('id', id)
+        .then((result) => {
+          res.status(200).send({ 'status': 'ok' });
+        })
+        .catch((err) => {
+          res.status(500).send(err);
+          logger.main.error(err)
+        })
+      logger.main.debug(util.format('%o', req.body || 'request body empty'));
+    } else {
+      res.status(500).json({ 'error': 'User ID 1 is protected' });
```

Comment:
```suggestion
      res.status(400).json({ 'error': 'User ID 1 is protected' });
```

400 error - the user did a bad, not the server

Reply:
there'll be lots of these scattered around i reckon. i was copy and pasting between routes for these... I'll go thru and check them all :)